tensorflow/python/keras/regularizers_test.py

```diff
@@ -75,9 +87,49 @@ def test_activity_regularization(self):
 
   def test_zero_regularization(self):
     inputs = keras.backend.ones(shape=(10, 10))
-    layer = keras.layers.Dense(3, kernel_regularizer=keras.regularizers.l2(0))
+    layer = keras.layers.Dense(3, kernel_regularizer=regularizers.l2(0))
     layer(inputs)
 
+  def test_regularization_shared_layer(self):
+    with self.cached_session():
```

Comment:
Rather than using `cached_session`, use the decorator `run_all_keras_modes` (see usage examples elsewhere). You'll need to pass the `run_eagerly` arg in compile (again, see usage examples for details). This provides greater coverage.